frog/imports/client/Wiki/index.js

```diff
@@ -140,9 +141,10 @@ class WikiComp extends React.Component<WikiCompPropsT, WikiCompStateT> {
 
     window.wikiDoc = this.wikiDoc;
     if (!this.props.embed) {
-      Mousetrap.bindGlobal('ctrl+n', () =>
-        this.setState({ createModalOpen: true })
-      );
+      Mousetrap.bindGlobal('ctrl+n', () => {
+        if (!this.state.settings?.noNewPage && !this.state.isOwner)
```

Comment:
3\*2\*3 = 18 conditional possibilities, enforcing the presence of `settings` or making `isOwner` defined would help reduce this number

Reply:
`isOwner` is now defined but cannot strictly enforce the presence of `settings` yet unless we can upgrade the existing wikis in production with the settings field in the database

Reply:
We could add a default value for settings on the client side to at least clarify the default behaviour, would remove a number of `?` notations and improve the confidence in the code